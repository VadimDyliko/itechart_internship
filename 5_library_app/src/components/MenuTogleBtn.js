import React from 'react';
import classnames from 'classnames';

const MenuTogleBtn = ({isOpenHandler, isOpen}) => {

  let btnLine = classnames({
    "btn__line hide": isOpen,
    "btn__line": !isOpen
  })
  let btnLineFirst = classnames({
    "btn__line btn__line_first": isOpen,
    "btn__line": !isOpen
  })
  let btnLineSecond = classnames({
    "btn__line btn__line_second": isOpen,
    "btn__line": !isOpen
  })

  const clickHandler = () => {
    isOpenHandler()
  }

  return (
    <button className="nav-bar__btn" onClick={clickHandler}>
      <div className={btnLineFirst}></div>
      <div className={btnLineSecond}></div>
      <div className={btnLine}></div>
    </button>
  )
}

export default MenuTogleBtn
