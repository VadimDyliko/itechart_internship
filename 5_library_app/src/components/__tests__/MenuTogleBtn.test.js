import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

import React from 'react';
import ReactDOM from 'react-dom';
import MenuTogleBtn from '../MenuTogleBtn'
import renderer from 'react-test-renderer';

describe('MenuTogleBtn render while SideBar is close', () => {
  let isOpenHandlerMock = jest.fn();
  let props = {
    isOpen: false,
    isOpenHandler: isOpenHandlerMock
  }
  let component = shallow(<MenuTogleBtn {...props}/>);

  it('renders properly', () => {
    expect(component).toMatchSnapshot()
  })

  it('onClick event should call isOpenHandler', () => {
    component.simulate('click')
    expect(isOpenHandlerMock).toHaveBeenCalledTimes(1)
  })
})

describe('MenuTogleBtn render while SideBar is open', () => {
  let props = {
    isOpen: true
  }
  let component = shallow(<MenuTogleBtn {...props}/>);

  it('renders properly', () => {
    expect(component).toMatchSnapshot()
  })
})
