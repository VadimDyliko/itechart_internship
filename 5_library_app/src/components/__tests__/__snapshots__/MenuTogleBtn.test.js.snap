// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuTogleBtn render while SideBar is close renders properly 1`] = `ShallowWrapper {}`;

exports[`MenuTogleBtn render while SideBar is open renders properly 1`] = `ShallowWrapper {}`;
